packages/docusaurus-plugin-content-blog/src/index.ts

```diff
@@ -143,7 +148,10 @@ export default function pluginContentBlog(
       // Blog pagination routes.
       // Example: `/blog`, `/blog/page/1`, `/blog/page/2`
       const totalCount = blogPosts.length;
-      const numberOfPages = Math.ceil(totalCount / postsPerPage);
+      const postsPerPage =
+        postsPerPageOption === 'ALL' ? totalCount : postsPerPageOption;
+      const numberOfPages =
+        postsPerPage === 0 ? 1 : Math.ceil(totalCount / postsPerPage);
```

Comment:
It's not clear why we need this? In which case is `postsPerPage  === 0`?

Reply:
When `postsPerPageOption === 'ALL' && totalCount === 0`. It's a niche case but guarding against it doesn't complicate things anyways

Reply:
There's an early return for when there's no blog post already so total count can't be 0.

I'd prefer to avoid introducing code that would never run ;)

Reply:
Ah, didn't notice that. Thanks for pointing out